sys/kern/vm_map.c

```diff
@@ -161,6 +161,34 @@ void vm_segment_destroy(vm_map_t *map, vm_segment_t *seg) {
   vm_segment_free(seg);
 }
 
+void vm_segment_destroy_range(vm_map_t *map, vm_segment_t *seg, vaddr_t start,
+                              vaddr_t end) {
+  assert(mtx_owned(&map->mtx));
+
+  if (seg->start == start && seg->end == end) {
+    vm_segment_destroy(map, seg);
+    return;
+  }
+
+  size_t length = end - start;
+  vm_object_remove_range(seg->object, start - seg->start, length);
+  pmap_remove(map->pmap, start, end);
+
+  if (seg->start == start) {
+    seg->start = end;
+  } else if (seg->end == end) {
+    seg->end = start;
+  } else { /* a hole inside the segment */
+    vm_object_t *obj = vm_object_clone(seg->object);
+    vm_object_remove_range(obj, 0, end - seg->start);
```

Comment:
I think you should remove range of length `start - seg->start`.

The range between `start` and `end` was removed in line 174.

Reply:
Oh, yes, sorry, my fault, I fixed it.